src/EventListener/InitializeSystemListener.php

```diff
@@ -0,0 +1,257 @@
+        // They will be removed in Contao 5 and should not be used anymore.
+        define('TL_MODE', $mode);
+        define('TL_START', microtime(true));
+        define('TL_ROOT', dirname($this->rootDir));
+        define('TL_REFERER_ID', substr(md5(TL_START), 0, 8));
+        define('TL_SCRIPT', $route);
+
+        // Define the login status constants in the back end (see #4099, #5279)
+        if ('BE' === TL_MODE) {
+            define('BE_USER_LOGGED_IN', false);
+            define('FE_USER_LOGGED_IN', false);
+        }
+
+        require __DIR__ . '/../../contao/helper/functions.php';
+        require __DIR__ . '/../../contao/config/constants.php';
+        require __DIR__ . '/../../contao/helper/interface.php';
+        require __DIR__ . '/../../contao/helper/exception.php';
+
+        // Try to disable the PHPSESSID
+        @ini_set('session.use_trans_sid', 0);
+        @ini_set('session.cookie_httponly', true);
+
+        // Set the error and exception handler
+        @set_error_handler('__error');
+        @set_exception_handler('__exception');
+
+        // Log PHP errors
+        @ini_set('error_log', TL_ROOT . '/system/logs/error.log');
+
+        // Include the Config class if it does not yet exist
+        if (!class_exists('Config', false)) {
+            require_once __DIR__ . '/../../contao/library/Contao/Config.php';
+            class_alias('Contao\\Config', 'Config');
+        }
+
+        // Include some classes required for further processing
+        require_once __DIR__ . '/../../contao/library/Contao/ClassLoader.php';
+        class_alias('Contao\\ClassLoader', 'ClassLoader');
+
+        require_once __DIR__ . '/../../contao/library/Contao/TemplateLoader.php';
+        class_alias('Contao\\TemplateLoader', 'TemplateLoader');
+
+        require_once __DIR__ . '/../../contao/library/Contao/ModuleLoader.php';
+        class_alias('Contao\\ModuleLoader', 'ModuleLoader');
+
+        // Preload the configuration (see #5872)
+        Config::preload();
+
+        // Try to load the modules
+        try {
+            ClassLoader::scanAndRegister();
+        } catch (\UnresolvableDependenciesException $e) {
+            die($e->getMessage()); // see #6343
+        }
+
+        // Override the SwiftMailer defaults
+        \Swift::init(function() {
+            $preferences = \Swift_Preferences::getInstance();
+            $preferences->setTempDir(TL_ROOT . '/system/tmp')->setCacheType('disk');
+            $preferences->setCharset(Config::get('characterSet'));
+        });
+
+        // Define the relative path to the installation (see #5339)
+        if (Config::has('websitePath') && TL_SCRIPT /* FIXME */ != 'contao/install.php') {
+            Environment::set('path', Config::get('websitePath'));
+        } elseif ('BE' === TL_MODE) {
+            Environment::set('path', preg_replace('/\/contao\/[a-z]+\.php$/i', '', Environment::get('scriptName')));
```

Comment:
Right here, the preg_replace will never match anymore as no URL will be `/contao/xyz.php` anymore but only `/contao/xyz`.